planus/src/builder.rs

```diff
@@ -158,6 +174,18 @@ impl Builder {
         self.inner.len() + vtable_size + 4 + needed_padding + object_size + 4
     }
 
+    /// Finish writing the internal buffer and return a byte slice of it.
+    ///
+    /// This will make sure all alignment requirements are fullfilled and that
+    /// the file identifier has been written if specified.
+    ///
+    /// It can also be used to directly serialize an owned flatbuffers struct:
+    /// ```
+    /// use planus::Builder;
+    /// use planus_example::monster_generated::my_game::sample::Weapon;
+    /// let mut builder = Builder::new();
+    /// let data = builder.finish(Weapon { name: Some("Sword".to_string()), damage: 12 }, None);
```

Comment:
Suggestion: would it make sense to incorporate the fact the it could take a reference instead of consuming the object?

Reply:
I like your added example, but I was thinking of an example where you serialize a reference to an owned object (i.e. you have an owned object but you don't want to drop it as you serialize)